File: alice.py







import time
from openvpn_api import Client
import ssl
import datetime

# Lista de configuraciones de VPN (dirección IP, puerto, etc.)
vpn_configs = [
    {"ip": "vpn_server1_ip", "port": 5555},
    {"ip": "vpn_server2_ip", "port": 5555},
    # Agrega más configuraciones de VPN según sea necesario
]

# Función para conectar al servidor VPN especificado
def connect_to_vpn(vpn_config):
    vpn_client = Client()
    try:
        vpn_client.connect(vpn_config["ip"], vpn_config["port"])
        return vpn_client
    except Exception as e:
        print(f"No se pudo conectar al servidor VPN {vpn_config['ip']}: {e}")
        return None

# Función para verificar la caducidad del certificado del servidor VPN
def check_cert_expiry(server_ip):
    cert = ssl.get_server_certificate((server_ip, 443))
    x509 = ssl.PEM_cert_to_DER_cert(cert)
    x509 = ssl.DER_cert_to_PEM_cert(x509)
    x509 = ssl.load_cert_string(x509)
    expiry_date = x509.get_notAfter()
    expiry_date = datetime.datetime.strptime(expiry_date.decode('ascii'), '%Y%m%d%H%M%SZ')
    days_left = expiry_date - datetime.datetime.now()
    return days_left.days

# Bucle principal para mantener la conexión VPN activa y verificar la caducidad del certificado
current_vpn = 0
while True:
    vpn_client = connect_to_vpn(vpn_configs[current_vpn])
    
    # Si no se pudo conectar, intenta con el siguiente servidor VPN en la lista
    if vpn_client is None:
        current_vpn = (current_vpn + 1) % len(vpn_configs)
        continue
    
    print(f"Conectado a VPN {current_vpn + 1}")
    
    # Verificar la caducidad del certificado cada día
    days_left = check_cert_expiry(vpn_configs[current_vpn]["ip"])
    if days_left < 30:
        print(f"¡Atención! El certificado del servidor VPN {current_vpn + 1} caducará en menos de 30 días.")
    
    # Esperar un tiempo antes de cambiar a la siguiente VPN
    time.sleep(3600)  # Esperar una hora antes de volver a verificar
    
    # Desconectar la VPN actual
    vpn_client.disconnect()
    
    # Cambiar al siguiente servidor VPN en la lista
    current_vpn = (current_vpn + 1) % len(vpn_configs)

File: armynet/Alice.c









#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include <openssl/ssl.h>
#include <openssl/x509.h>
#include <openssl/pem.h>

// Función para conectar al servidor VPN especificado
void connect_to_vpn(const char* ip, int port) {
    // Aquí iría el código para establecer la conexión VPN
    printf("Conectado a VPN en %s:%d\n", ip, port);
}

// Función para verificar la caducidad del certificado del servidor VPN
int check_cert_expiry(const char* server_ip) {
    // Aquí iría el código para obtener y verificar el certificado del servidor VPN
    // Retornar el número de días restantes hasta la caducidad del certificado
    return 30; // Supongamos que siempre quedan 30 días para la caducidad (simulación)
}

int main() {
    // Lista de configuraciones de VPN (dirección IP, puerto, etc.)
    const char* vpn_configs[][2] = {
        {"vpn_server1_ip", "5555"},
        {"vpn_server2_ip", "5555"}
        // Agrega más configuraciones de VPN según sea necesario
    };

    int num_vpn_configs = sizeof(vpn_configs) / sizeof(vpn_configs[0]);

    // Bucle principal para mantener la conexión VPN activa y verificar la caducidad del certificado
    int current_vpn = 0;
    while (1) {
        const char* ip = vpn_configs[current_vpn][0];
        int port = atoi(vpn_configs[current_vpn][1]);

        // Conectar al servidor VPN actual
        connect_to_vpn(ip, port);

        // Verificar la caducidad del certificado del servidor VPN actual
        int days_left = check_cert_expiry(ip);
        if (days_left < 30) {
            printf("¡Atención! El certificado del servidor VPN %s caducará en menos de 30 días.\n", ip);
        }

        // Esperar un tiempo antes de cambiar al siguiente servidor VPN
        sleep(3600);  // Esperar una hora antes de volver a verificar

        // Cambiar al siguiente servidor VPN en la lista
        current_vpn = (current_vpn + 1) % num_vpn_configs;
    }

    return 0;
}